libazureinit/src/imds.rs

```diff
@@ -87,65 +90,110 @@ where
     }
 }
 
-pub async fn query(client: &Client) -> Result<InstanceMetadata, Error> {
-    let url = "http://169.254.169.254/metadata/instance?api-version=2021-02-01";
+// Set of StatusCodes that should be retried,
+// e.g. 400, 404, 410, 429, 500, 503.
+const RETRY_CODES: &[StatusCode] = &[
+    StatusCode::BAD_REQUEST,
+    StatusCode::NOT_FOUND,
+    StatusCode::GONE,
+    StatusCode::TOO_MANY_REQUESTS,
+    StatusCode::INTERNAL_SERVER_ERROR,
+    StatusCode::SERVICE_UNAVAILABLE,
+];
+
+static DEFAULT_IMDS_URL: &str =
+    "http://169.254.169.254/metadata/instance?api-version=2021-02-01";
+
+pub async fn query(
+    client: &Client,
+    retry_interval: Duration,
+    total_timeout: Duration,
+    inputurl: Option<&str>,
```

Comment:
NIT: we should rename this to `input_url` for consistency

Reply:
Done